### DISCUSSION TITLE
CoreWCF.Kafka

### DISCUSSION THREAD

**Discussion by nf17** [CoreWCF/CoreWCF#1241] [General]
Are there any examples around how to use CoreWCF.Kafka? I am working on a dotnet Kafka consumer/subscriber and i am using a combination of ASP.net with BackgroundService to listen to Kafka events. Was curious how this problem is solved in CoreWCF. Appreciate if anyone point me in the right direction.

**Comment by g7ed6e** [comment]
> Are there any examples around how to use CoreWCF.Kafka? I am working on a dotnet Kafka consumer/subscriber and i am using a combination of ASP.net with BackgroundService to listen to Kafka events. Was curious how this problem is solved in CoreWCF. Appreciate if anyone point me in the right direction.

Hello @nf17 for the time being you can refer to unit tests to read sample code. Please hava a look to this basic scenario. https://github.com/CoreWCF/CoreWCF/blob/b75461acd825614ab0bcfe4326684efa0581560b/src/CoreWCF.Kafka/tests/BasicServiceTests.cs#L68C8-L68C8

Regarding the CoreWCF implementation, we are using an `IHostedService` too.
